### PR TITLE
drivers/pinmux: stm32: Remove deprecated pinctrl definitions

### PR DESCRIPTION
**Pull request by erwango**
STM32 pinctrl macros were set as deprecated following integration of stm32 pinctrl in device tree.
This was the case since 1 release now, so let's remove them and make appropriate related changes

### CODE REVIEW

File: boards/arm/stm32g0316_disco/pinmux.c [hunk 1]
```diff
@@ -9,8 +9,6 @@
 #include <stm32_ll_bus.h>
 #include <stm32_ll_system.h>
 
-#include <pinmux/stm32/pinmux_stm32.h>
```
Comment (ABOSTM) [inline boards/arm/stm32g0316_disco/pinmux.c:12]: To be replaced by `#include <pinmux/pinmux_stm32.h>` ?
Or no more necessary ?

Comment (erwango) [reply boards/arm/stm32g0316_disco/pinmux.c:12]: From what I see, it doesn't need this header anymore.

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Can you send a follow up PR to update the release notes "removal" section.

**Pull request merged**